### ISSUE TITLE
Worker needs to be refactored.

### ISSUE THREAD

**Issue by cmeiklejohn** [drip/drip#15]
Worker needs to be broken out into modules and refactored to be more readable.  This work should be based off of the coffeescript branch.

**Comment by barnabyc** [comment]
I completely agree. Do you feel this is required for opensourceification?

**Comment by cmeiklejohn** [comment]
Yes.  I'd like it to be cleaned up so we can more easily debug what is happening and dynamically scale workers if necessary.

**Comment by cmeiklejohn** [comment]
Reassigned to me, I'll take a pass at it over the next few days.

**Comment by hojberg** [comment]
is it open/close bingo in here?
